servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -1049,8 +1058,9 @@ boolean shouldRetryOperation(ResourceException ex, int retryCount) {
         switch (ex.getCode()) {
 
             case ResourceException.CONFLICT:
-
-                retry = true;
+                if (!ex.getMessage().contains("already has a pending request in a different state")) {
```

Comment:
we should not make our logic based on message content. it is possible that the customer might write their own datastore plugin and may not be aware that they must return something specific in the error message. instead of returning CONFLICT, we should just return BAD_REQUEST and the error message can indicate what we already have.